### DISCUSSION TITLE
using this within a docker container?

### DISCUSSION THREAD

**Discussion by 3dmation** [grodansparadis/node-red-contrib-socketcan#15] [Q&A] [answered]
I'm able to bring this in, and set my 'channel' to 'can0'  This is the canbus i'm using with my usb>can usb device.  I can see the canbus running with a candump and also see the device i have reporting into the dump.  however, my node-red won't report anything as I get an error from the socketcan-out node.  It's "Error: Error while creating channelcan0".  I believe this is happening because it doesn't see the can0 bus since it's in a container.  Do you know how we can pass this through the container?  perhaps with the docker-compose.yaml file?

**Comment by grodansparadis** [comment, marked as answer]
Hi,
no sorry, I don't use docker or similar so I can't tell. As far as I can tell this should be a quite common use case but googling don't give much either.  Does this [](https://chemnitzer.linux-tage.de/2021/media/programm/folien/210.pdf )help?

**Reply by 3dmation** [reply]
At first, I thought this wasn't helpful.  However, docker does have strange issues especially when you try to use anything for that app that is outside the container.  Instead of fighting it, i tried just not using it within docker and it works just fine.  Thank you for the prodding.

**Comment by grodansparadis** [comment]
Good to hear. Cheers /Ake
